txnkv/transaction/2pc.go

```diff
@@ -179,6 +179,9 @@ type twoPhaseCommitter struct {
 
 	// The total number of kv request after batch split.
 	prewriteTotalReqNum int
+
+	// assertion error happened when initializing mutations, could be false positive if pessimistic lock is lost
+	assertionErr error
```

Comment:
It seems that the `twoPhaseCommitter.execute` handles assertion errors from everywhere. How about renaming this to something like `stashedAssertionError`?